### PR TITLE
Disc number sorting (#331)

### PR DESCRIPTION
**Pull request by rickosborne**
I've only done rudimentary testing, but it seems to work without crashing.  I'm also not sure if I got all the right places?

- Adds `disc_number` database column
- Update Track.compareTo to try to use disc number, if present
- Update TracksActivity.refreshTracks to try to use disc number, if present

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/musicplayer/extensions/Context.kt [hunk 1]
```diff
@@ -185,6 +186,10 @@ fun Context.getAlbumTracksSync(albumId: Long): ArrayList<Track> {
         projection.add(Audio.Media.BUCKET_DISPLAY_NAME)
     }
 
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/musicplayer/extensions/Context.kt:189]: use helper functions like isRPlus instead of these please

Comment (rickosborne) [reply app/src/main/kotlin/com/simplemobiletools/musicplayer/extensions/Context.kt]: Done.

File: app/src/main/kotlin/com/simplemobiletools/musicplayer/extensions/Context.kt [hunk 2]
```diff
@@ -204,8 +210,13 @@ fun Context.getAlbumTracksSync(albumId: Long): ArrayList<Track> {
         } else {
             ""
         }
+        val discNumber = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/musicplayer/extensions/Context.kt:213]: use isRPlus here too

File: app/src/main/kotlin/com/simplemobiletools/musicplayer/models/Track.kt [hunk 3]
```diff
@@ -52,6 +53,9 @@ data class Track(
             }
             sorting and PLAYER_SORT_BY_TRACK_ID != 0 -> {
                 when {
+                    discNumber <= 0 && other.discNumber > 0 -> 1
+                    discNumber > 0 && other.discNumber <= 0 -> -1
+                    discNumber != other.discNumber -> AlphanumericComparator().compare(discNumber.toString(), other.discNumber.toString())
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/musicplayer/models/Track.kt:58]: no need to use alphanumeric comparator if the value is an int, it is needed at strings only. Guess we dont need it a couple lines lower at "else" with comparing trackId either.

**Pull request closed without merging**